frontend/packages/console-shared/src/test-utils/utils.ts

```diff
@@ -157,3 +157,25 @@ export function searchYAML(needle: string, name: string, namespace: string, kind
   const result = execSync(`kubectl get -o yaml -n ${namespace} ${kind} ${name}`).toString();
   return result.search(needle) >= 0;
 }
+
+type CardInfo = {
+  text: string;
+  index: number;
+};
+
+const getOperatorHubCards = async (): Promise<CardInfo[]> => {
+  const selector = $$('article .co-resource-item__resource-name');
+  const cards: CardInfo[] = await selector.map(async (elem, index) => {
+    return {
+      text: await elem.getText(),
+      index,
+    };
+  });
+  return cards;
+};
+
+export const getOperatorHubCardIndex = async (name: string): Promise<CardInfo['index']> => {
```

Comment:
Wouldn't `element(by.cssContainingText('article a', name));` be sufficient?
It seems to me like exactly what you're looking for.

Reply:
Discussed over private channel. Not sufficient due to `article a` element having the same text in all cards.